src/ast/mod.rs

```diff
@@ -102,10 +102,26 @@ fn write_tree_view_recursive<'arena, Node>(
     }
 }
 
+/// The specification of an NodeClass that Class can have
+pub trait NodeClass<'arena>: Copy + std::fmt::Debug + Eq + std::hash::Hash {
```

Comment:
I think a better doc-comment would be
```rust
/// All the possible types which [`Ast`] nodes can take.
```

Also I think that `NodeClass` should probably be called `AstClass` or just `Class` (both would be consistent with the naming of `Ast`).